inject.go

```diff
@@ -49,7 +49,7 @@ type TypeMapper interface {
 	Set(reflect.Type, reflect.Value) TypeMapper
 	// Returns the Value that is mapped to the current type. Returns a zeroed Value if
 	// the Type has not been mapped.
```

Comment:
I'd add a sentence to the comment indicating that it returns an error when two implementations are available.